Preprocessing/Base/converter.py

```diff
@@ -217,85 +216,96 @@ def split_into_time_windows(signals: mne.io.RawArray, sample_frequency: int, sec
+        filledLength = int(length * iteration // total)
+        bar = fill * filledLength + '-' * (length - filledLength)
+        print(f'\r{prefix} |{bar}| {percent}% {suffix}', end=printEnd)
+
+    # Initial Call
+    print_progress_bar(0)
+    # Update Progress Bar
+    for i, item in enumerate(iterable):
+        yield item
+        print_progress_bar(i + 1)
+    # Print New Line on Complete
+    print()
+
+
+def process_folder(folder_name: str, mode: Literal['Edf', 'Csv', 'Eea'],
+                   patient_state: Literal['health', 'schizophrenia'], output_folder: str = None) -> None:
     """
     Main function. Read all files from provided folder and process them.
     If output_folder is not provided, it will use default one.
 
+    :param patient_state: Information about the state of patients processed. Needed for file save.
     :param output_folder: Folder where data should be saved.
     :param folder_name: Folder which stores all the files
     :param mode: File extension which are in the folder. THERE SHOULDN'T BE MORE THAN ONE FILE TYPE IN FOLDER
     """
     file_names = os.listdir(folder_name)
+    set_log_level("Critical")
+    object_dump = []
 
-    if mode == 'edf':
-        for file_name in file_names:
+    if mode == 'Edf':
+        for file_name in progress_bar(file_names, prefix='Progress: '):
             data = get_signals_from_edf(os.path.join(folder_name, file_name))
-            preprocess_data_all_steps(data, file_name[:file_name.index('.')],
-                                      output_folder if output_folder is not None else 'EdfData')
-    elif mode == 'csv':
-        for file_name in file_names:
+            processed_data = preprocess_data_all_steps(data)
+            object_dump.append(processed_data)
+    elif mode == 'Csv':
+        for file_name in progress_bar(file_names, prefix='Progress: '):
             data = get_signals_from_csv(os.path.join(folder_name, file_name))
             for index, signal_chunk in enumerate(data):
-                file_name_no_extension = file_name[:file_name.index('.')]
-                preprocess_data_all_steps(signal_chunk, f'{file_name_no_extension}_chunk_{index}',
-                                          output_folder if output_folder is not None else 'CsvData')
-                print(f'Processed {index}')
-    elif mode == 'eea':
-        for file_name in file_names:
+                processed_data = preprocess_data_all_steps(signal_chunk)
+                object_dump.append(processed_data)
+    elif mode == 'Eea':
+        for file_name in progress_bar(file_names, prefix='Progress: '):
             data = get_signals_from_eea(os.path.join(folder_name, file_name))
-            preprocess_data_all_steps(data, file_name[:file_name.index('.')],
-                                      output_folder if output_folder is not None else 'EeaData')
+            processed_data = preprocess_data_all_steps(data)
```

Comment:
Poproszę Cię jeszcze o zmianę formatu tablicy słowników na:
`{ "id": "file_name", "eeg": { "F8": "2323...", "O2": "234145...", ... } }`
gdzie file_name to nazwa pliku bez rozszerzenia: `file_name_no_extension`

Reply:
Przygotowanie datasetów jest prawie zrobione ( pliki są pobierane, muszę jeszcze rozpakowywać pliki do folderów ale na ten moment nie zdążyłem tego zrobić). Do tego dodałem CLI dla preprocessingu aby można było przetworzyć wszystkie pliki. Pliki csv są w formacie `{id: chunk_{chunk_index}_{file_name}, eeg:...}` reszta jest normalnie. Readme też jest zrobione

Reply:
Co ważne jednak wciąż przyjąłem flow że w jednym folderze może być tylko jeden stan pacjenta (albo chory albo zdrowy). Rozpoznawanie tego wymagałoby przebudowania działania całego wczytywania plików w preprocessingu. Mam nadzieje że rozbicie tego na osobne pliki .pk jest okej

Reply:
Do tego zakładam działanie przygotowania datasetów w taki sposób: Każdy dataset pobieram i potem dziele na osobne foldery (csv_ill, csv_healthy, eeg_ill itd.) Mogę też od razu przerzucić wszystko do odpowiednio folderów ill oraz healthy ale może się robić burdel przez to więc imo przerzucanie plików do odpowiednich folderów zostawiłbym osobom które będą działać na tym projekcie